ddsketch/ddsketch.go

```diff
@@ -248,3 +248,9 @@ func changeStoreMapping(oldMapping, newMapping mapping.IndexMapping, oldStore, n
 		return false
 	})
 }
+
+func (s *DDSketch) Reweight(w float64) {
+	s.zeroCount *= w
+	s.positiveValueStore.Reweight(w)
```

Comment:
The error is ignored it seems. Also, should we check `w` here for `<= 0` to avoid altering the store in case of invalid factor? And maybe handle the `==1` case here as well?

Reply:
Good catch, thanks. Updated.